File: lib-esphome-client/src/main/java/io/github/arhor/esphome/client/async/EspHomeEncryptionConfig.java
package io.github.arhor.esphome.client.async;

public record EspHomeEncryptionConfig(
    boolean enabled,
    String key
) {
    public EspHomeEncryptionConfig {
        if (enabled && (key == null || key.isBlank())) {
            throw new IllegalArgumentException("key must be configured when encryption is enabled");
        }
    }

    public static EspHomeEncryptionConfig disabled() {
        return new EspHomeEncryptionConfig(false, null);
    }
}

File: services/lib-esphome-client/src/main/kotlin/io/github/arhor/esphome/client/config/EspHomeEncryptionConfig.kt
package io.github.arhor.esphome.client.config

data class EspHomeEncryptionConfig(
    val enabled: Boolean = false,
    val key: String? = null,
) {
    init {
        require(!enabled || !key.isNullOrBlank()) {
            "key must be configured when encryption is enabled"
        }
    }
}
